### ISSUE TITLE
Soft Delete for Users

### ISSUE THREAD

**Issue by ghost** [thedevdojo/voyager#4106]
## Version information

 - Laravel: v5.7
 - Voyager: v1.1
 - PHP: 7.2
 - Database:MySQL 8.0

## Description of problem
I have a system which includes a lot of user. A user has a project, if the admin deleted the user from User's table, project created and user should still be indicated as author. Already added `deleted_at` on the users table and added soft delete to User Model, still no luck.

```
use Illuminate\Database\Eloquent\SoftDeletes;

class User extends \TCG\Voyager\Models\User
{
	use HasApiTokens, Notifiable, SoftDeletes;
	
	protected $dates = ['deleted_at'];
```

**Comment by MrCrayon** [comment]
Did you add withTrashed() to belongsTo relationship?

**Comment by ghost** [comment]
> Did you add withTrashed() to belongsTo relationship?

@MrCrayon 

I didn't. Is it needed to add ?
btw, It's the default user model I'm using .

**Comment by MrCrayon** [comment]
As per Laravel documentation, yes. 
https://laravel.com/docs/5.8/eloquent#soft-deleting

**Comment by ghost** [comment]
@MrCrayon 

Sorry sir, but I think you're talking about retrieving the trashed data.
My problem is the data is not being soft deleted.

**Comment by MrCrayon** [comment]
Maybe rephrase your issue because that's what it sounds like, if user is deleted and not soft deleted then that's the problem I would not even mention any other relationship where the user doesn't show up :wink: 

Also if this is the problem it should be marked as possible BUG not feature.

Anyway I just tried on a fresh install and it works, are you sure Voyager is set to use your User Model and not the default one?

Try checking in BREAD -> users -> Model Name

**Comment by ghost** [comment]
@MrCrayon 

I tried using fresh install. Model is correct.
Still, deleting user was not soft deleted.

**Comment by MrCrayon** [comment]
Do you see this in user list?
![Schermata da 2019-05-02 11-55-33](https://user-images.githubusercontent.com/6875243/57068194-51ed9480-6cd1-11e9-9286-8fe644fef498.png)
If you don't see the third button soft delete is not enabled.

Can you list everything you do after fresh install to enable soft delete?

**Comment by ghost** [comment]
@MrCrayon 

I'm not seeing that one. What did i missed ?

**Comment by ghost** [comment]
already added on User Model.
![image](https://user-images.githubusercontent.com/36871215/57117022-a2c1c380-6d8b-11e9-9d08-8e8f5bab974e.png)

**Comment by MrCrayon** [comment]
The only way I can explain it is if that class is not the one used.

Try adding route below in your route.php

```php
Route::group(['prefix' => 'admin'], function () {
    Voyager::routes();

    Route::get('/users', function () {
        return get_class(auth()->user());
    });
});
```

Then go to /admin/users you should see which class you are using

**Comment by MrCrayon** [comment]
Nevermind what's written above is going to give you App\User.

The problem is VoyagerBaseController uses what's in database data_types.model_name and that's, by default, set as TCG\Voyager\Models\User, so unless you changed it that's the model used for users as BREAD.

BTW 

> Try checking in BREAD -> users -> Model Name

That would have already solved your problem yesterday :wink:

**Comment by ghost** [comment]
@MrCrayon 

Thank You! :)

**Comment by MrCrayon** [comment]
You are welcome!

Please close the issue if it has been solved.
